plugins/org.eclipse.n4js/src/org/eclipse/n4js/scoping/utils/ProjectImportEnablingScope.java

```diff
@@ -251,7 +256,12 @@ private IN4JSProject findProject(String projectId, IN4JSProject project) {
 		if (Objects.equals(project.getProjectId(), projectId)) {
 			return project;
 		}
-		for (IN4JSProject p : project.getDependencies()) {
+
+		Iterable<IN4JSProject> dependencies = TypeDefinitionsAwareDependenciesSupplier.get(project);
```

Comment:
Maybe we can make use of `IResourceScopeCache` so this comparatively expensive operation is only performed once per resource (at least in the scoping).

Reply:
Thanks! Yes, I introduced the `MultiCleartriggerCache` to take care of this.